src/main/java/manageezpz/model/AddressBook.java

```diff
@@ -44,6 +51,102 @@ public AddressBook(ReadOnlyAddressBook toBeCopied) {
         resetData(toBeCopied);
     }
 
+    /**
+     * Gets the list of all tasks.
+     * @return A string representing all tasks
+     */
+    public String listTask() {
+        Iterator<Task> taskIterators = tasks.iterator();
+        String result = "";
+
+        while (taskIterators.hasNext()) {
+            Task task = taskIterators.next();
+            result = String.join("\n", result, task.toString());
+        }
+
+        return result;
+    }
+
+    /**
+     * Gets the list of all tasks with the options
+     * @param option Option provided to filter list
+     * @return All tasks that satisfy the option given
+     */
+    public String listTask(Prefix option) {
+        if (PREFIX_TODO.equals(option)) {
+            return listTodo();
+        } else if (PREFIX_DEADLINE.equals(option)) {
+            return listDeadline();
+        } else if (PREFIX_EVENT.equals(option)) {
+            return listEvent();
+        } else if (PREFIX_TODAY.equals(option)) {
+            return listToday();
+        }
+        assert true : "Invalid option, should be checked in list command parser";
+        return null;
+    }
+
+    /**
+     * Returns the list of all todos.
+     * @return List of all todos.
+     */
+    private String listTodo() {
+        Iterator<Task> taskIterators = tasks.iterator();
+        String result = "";
+
+        while (taskIterators.hasNext()) {
+            Task task = taskIterators.next();
+            if (task instanceof Todo) {
+                result = String.join("\n", result, task.toString());
+            }
+        }
+
+        return result;
+    }
+
+    private String listDeadline() {
+        Iterator<Task> taskIterators = tasks.iterator();
+        String result = "";
+
+        while (taskIterators.hasNext()) {
+            Task task = taskIterators.next();
+            if (task instanceof Deadline) {
+                result = String.join("\n", result, task.toString());
+            }
+        }
+
+        return result;
+    }
+
+    private String listEvent() {
+        Iterator<Task> taskIterators = tasks.iterator();
+        String result = "";
+
+        while (taskIterators.hasNext()) {
+            Task task = taskIterators.next();
+            if (task instanceof Event) {
+                result = String.join("\n", result, task.toString());
+            }
+        }
+
+        return result;
+    }
+
+    private String listToday() {
+        Iterator<Task> taskIterators = tasks.iterator();
+        Date todayDate = Date.getTodayDate();
+        String result = "";
+
+        while (taskIterators.hasNext()) {
+            Task task = taskIterators.next();
+            if (task instanceof Deadline || task instanceof Event) {
+                result = String.join("\n", result, task.toString());
```

Comment:
For this, the todayDate is not used in adding to the result, should there be a usage?